paasta_tools/mesos_tools.py

```diff
@@ -678,9 +678,13 @@ def get_mesos_task_count_by_slave(mesos_state, slaves_list=None, pool=None):
     :returns: list of slave dicts {'task_count': SlaveTaskCount}
     """
     all_mesos_tasks = get_all_running_tasks()  # empty string = all app ids
-    slaves = {
-        slave['id']: {'count': 0, 'slave': slave, 'chronos_count': 0} for slave in mesos_state.get('slaves', [])
-    }
+    try:
+        slaves = {
+            slave['id']: {'count': 0, 'slave': slave, 'chronos_count': 0} for slave in mesos_state.get('slaves', [])
+        }
+    except SlaveDoesNotExist:
+        log.debug("Tried to get mesos slaves for id {}, but none existed.".format(mesos_state.get('id'), ''))
+        return {}
```

Comment:
Are you sure this is the right place to put this?

Just because one slave doesn't exist, doesn't mean we have to return an empty dictionary, right?

Aren't we throwing out the baby here?